### PR TITLE
Makefile and utilities

### PR DESCRIPTION
**Pull request by calvinatintel**

### CODE REVIEW

**Comment by eriknyquist** [comment]
Looks good, although I haven't tested yet. Will do this afternoon.

PS can you rebase that "Oops" commit out.....

**Comment by calvinatintel** [comment]
It's not ready at all

**Comment by calvinatintel** [comment]
`make setup` works on the first attempt. Following attempts fail because of `Firmware` and `X86-Samples` dirs. Need to take account of those.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+check-root:
+	@if [ `whoami` != root ]; then echo "Please run as sudoer/root" ; exit 1 ; fi
+
+install-dep: check-root
+	apt-get update
+	apt-get install -y git make gcc gcc-multilib g++ libc6-dev-i386 g++-multilib python3-ply python
+	cp -f $(M_DIR)/utils/drivers/rules.d/*.rules /etc/udev/rules.d/
+
+setup:
+	@if [ ! -d Firmware ] ; then git clone https://github.com/01org/CODK-M-Firmware.git	Firmware ; fi
```
Comment (calvinatintel) [inline Makefile]: Pretty ugly...

File: install-zephyr.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Z_PATH=$(readlink -m "$(pwd)/../zephyr")
+ZSDK_FILE="zephyr-sdk-${ZSDK_VER}-i686-setup.run"
+ZSDK_PATH=$(readlink -m "$(pwd)/../zephyr-sdk")
+
+if [ -d ${Z_PATH} ]; then
+    echo "Zephyr source already exists. Skipping installation."
+else
+    echo "Downloading Zephyr"
+    #curl -sL http://bit.ly/1Y6G8d2 | tar xz
+    curl -sL https://github.com/01org/zephyr/archive/codk-m.tar.gz | tar xz
```
Comment (calvinatintel) [inline install-zephyr.sh:13]: Downloading from our fork, so that we fetch the latest + Dino's changes

**Comment by calvinatintel** [comment]
Need to make it so that `make compile` compiles sketches. @bbaltz505 Or do we not want that?

**Comment by calvinatintel** [comment]
@bbaltz505 This is function but not very elegant. Ready for Alpha?

**Comment by calvinatintel** [comment]
I'll crush the commits into one if the PR looks good

**Comment by bbaltz505** [comment]
Looks good. Regarding your 'make compile' comment, I think we want to make sketch users spend a little effort to build a sketch so they move to C++.

**Comment by eriknyquist** [comment]
@calvinatintel  CODK applications are C++, but if they want to migrate sketches into CODK we are providing a tool for once-off conversion (which they must run themselves).

**Comment by calvinatintel** [comment]
@bbaltz505 @eriknyquist Commits crushed. Ready for merge?

**Comment by eriknyquist** [comment]
Yes, I think so. Let's get a README going for this tree too.

**Comment by calvinatintel** [comment]
Trying out a whole new approach based on A

**Comment by calvinatintel** [comment]
Everything works except for `upload-*-jtag`. I need to figure out how to use `if` statements within JTAG config files. Do we need JTAG for Alpha?

**Comment by eriknyquist** [comment]
Not sure. what are the if statements needed for?

**Comment by calvinatintel** [comment]
If flashing just `x86`, just `arc`, or both. A quick and dirty way is to duplicate config files and call the correct ones from `if` in bash

**Comment by eriknyquist** [comment]
and, which config file specifically

File: utils/scripts/flash-jtag.cfg [hunk 3]
```diff
@@ -0,0 +1,17 @@
+init
+reset halt
+targets 0
+clk32M 4000
+gdb_breakpoint_override hard
+
+source utils/scripts/partition.conf
+load_image   $arc_bin   $arc_start_addr
+load_image   $x86_bin   $quark_start_addr
```
Comment (calvinatintel) [inline utils/scripts/flash-jtag.cfg:9]: These throw error unless both `$arc_bin` and `$x86_bin` are defined. We need to `if` them out, or duplicate these with the variants, and use `if` from bash

**Comment by eriknyquist** [comment]
that's an openocd file right?

**Comment by calvinatintel** [comment]
yup

**Comment by eriknyquist** [comment]
Is there any way to pass parameters, or use env. vars in the config file? That way we could have one flashing script that takes a .bin file and start address as parameters

**Comment by eriknyquist** [comment]
oh but that still requires a script I guess, to run it with the right params

**Comment by calvinatintel** [comment]
Right. By the way, every comment is an email out to everyone. I hope everyone has a good filter setup.

**Comment by eriknyquist** [comment]
Hey, this is what code reviews are for. The "unsubscribe" button is right there! --->

**Comment by calvinatintel** [comment]
Well, I'll make the duplicate scripts tomorrow and get it "working". We can optimize later.

**Comment by bbaltz505** [comment]
Are we maintaining a partition.conf file for the memory addresses? Thunderdome builds used to provide this. I never saw it in Zephyr. For the JTAG config, can we just "build" it using a template and updating the lines we need with sed or something?

**Comment by calvinatintel** [comment]
@bigdinotech Do you know if Zephyr uses the same `partition.conf` as TD based firmware/software?

**Pull request merged**